nemo/collections/asr/models/label_models.py

```diff
@@ -337,6 +338,23 @@ def setup_finetune_model(self, model_config: DictConfig):
 
         logging.info(f"Changed decoder output to # {self.decoder._num_classes} classes.")
 
+    @torch.no_grad()
+    def get_embedding(self, path2audio_file):
+        audio, sr = librosa.load(path2audio_file, sr=None)
+        if sr != 16000:
+            audio = librosa.core.resample(audio, sr, 16000)
```

Comment:
The target sampling rate should be read from the model config. This will fail for a model that was trained with say 8 kHz data.

Reply:
I guess the whole idea is to not read from config, hence the quick way to get embeddings for a single audio file and thought about providing the target sample rate, but since all speaker models are 16Khz hence the 16000 number.

Reply:
No that doesn't mean the default should be 16khz. The models SR should be used, whatever it is, from it's own config. You can use self.cfg to access the config that was used to build the model c